WS2012R2/lisa/setupscripts/NET_Corruption.ps1

```diff
@@ -0,0 +1,248 @@
+{
+    "Error: testParams is missing the sshKey parameter"
+    return $False
+}
+
+if (-not (Test-Path ssh\${sshKey}))
+{
+    "Error: The SSH key 'ssh\${sshKey}' does not exist"
+    return $False
+}
+
+if (-not $ipv4)
+{
+    "Error: The ipv4 parameter was not provided by LISA"
+    return $False
+}
+
+$port = 1234
+$sourceFilePath = "/tmp/testfile"
+$destionationFilePath = ".\testfile"
+$netcatScriptPath = "listen.sh"
+$summaryLog = "${vmName}_summary.log"
+
+del $summaryLog -ErrorAction SilentlyContinue
+Write-Output "Covers: ${tcCovered}" | Tee-Object -Append -file $summaryLog
+
+"Info: Configuring VM"
+$cmd = "./NET_Corruption.sh ${sourceFilePath} ${port} ${netcatScriptPath} 2>/dev/null"
+$sts = bin\plink.exe -i ssh\${sshKey} root@${ipv4} "dos2unix NET_Corruption.sh 2>/dev/null"
+$sts = bin\plink.exe -i ssh\${sshKey} root@${ipv4} "chmod 755 NET_Corruption.sh 2>/dev/null"
+$sts = bin\plink.exe -i ssh\${sshKey} root@${ipv4} $cmd
+
+if (-not $?)
+{
+    $msg = bin\plink.exe -i ssh\${sshKey} root@${ipv4} "tail -n 1 summary.log"
+    $vm_log = bin\plink.exe -i ssh\${sshKey} root@${ipv4} "tail -n+2 summary.log" | %{$_.Split("`n")}
+    $vm_log = $vm_log -join "`n"
+    "${vm_log}"
+    Write-Output $msg | Tee-Object -Append -file $summaryLog
+    return $False
+}
+
+"Info: Checking system logs path"
+bin\plink.exe -i ssh\${sshKey} root@${ipv4} "[[ -f /var/log/syslog ]]"
+if (-not $?)
+{
+    $logPath = '/var/log/messages'
+}
+else
+{
+    $logPath = '/var/log/syslog'
+}
+
+"Info: Starting netcat server on VM Job"
+$cmd = "`"" + "setsid ./${netcatScriptPath} >/dev/null 2>&1 < /dev/null &" + "`""
+"Info: Running command ${cmd}"
+bin\plink.exe -i ssh\${sshKey} root@${ipv4}  $cmd
+
+$jobName = "ReceiveJob"
+$ipAddr = (Get-VMNetworkAdapter -VMName ${vmName})[1].IPAddresses[0]
+$cmd = "cmd.exe /C " + "'" + "bin\nc.exe -v -w 2 ${ipAddr} ${port} > ${destionationFilePath}" + "'"
+"Info: Running command ${cmd}"
+$cmd | Out-File ./nccmd.ps1
+$sts = ./nccmd.ps1
+Start-Job -Name ${jobName} -ScriptBlock {./nccmd.ps1}
+
+
+"Info: Checking for call traces in ${logPath}"
+$grepCmd = "grep -i 'Call Trace' ${logPath}"
+while ((Get-Job -Name ${jobName}).State -eq "Running")
+{
+    bin\plink.exe -i ssh\${sshKey} root@${ipv4} $grepCmd
+    if ($?)
+    {
+        $msg = "Error: Call traces found in ${logPath}"
+        Write-Output $msg | Tee-Object -Append -file $summaryLog
+        return $False
+    }
+}
+
+Get-Job -Name ${jobName} | Remove-Job
+
+"Info: Comparing hashes"
+$localHash = (Get-FileHash -Algorithm MD5 $destionationFilePath).Hash
+$hashCmd = "md5sum ${sourceFilePath}"
+$remoteHash = (bin\plink.exe -i ssh\${sshKey} root@${ipv4} $hashCmd).Split(" ")[0]
+
+if (-not $?)
+{
+    $msg = "Error: Unable to get file hash from VM"
+    Write-Output $msg | Tee-Object -Append -file $summaryLog
+    return $False
+}
+
+"Info : File hashes: ${remoteHash} - ${localHash}"
+if ($remoteHash.ToUpper() -ne $localHash) {
+    $msg = "Error: File hashes do not match."
```

Comment:
Netcat binary was missing from \bin folder and no error was raised for this case. I've added a check, at the beginning of the test script, to verify if the path ".\bin\nc.exe" actually exists.